.github/workflows/smoke-test.yaml

```diff
@@ -0,0 +1,186 @@
+# This workflow downloads artifacts from a (by default, draft) release and runs
+# a short smoke test where the application is installed and run and immediately
+# shut down.
+# Since we need contents-write permissions to look at draft releases, we
+# actually download the artifacts in a smaller job, then upload them into the
+# run and download it _again_ in the second (per-platform) job where no
+# permissions are required.
+name: Release smoke test
+permissions: {}
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: >
+          Download artifacts from release with this tag, rather than picking the
+          first draft release.
+        type: string
+
+jobs:
+  download-artifacts:
+    name: Find release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Needed to list draft releases
+    env:
+      RELEASE_TAG: ${{ inputs.tag }}
+    steps:
+    - name: Find release
+      if : inputs.tag == ''
+      run: >-
+        set -o xtrace;
+        printf "RELEASE_TAG=%s\n" >>"$GITHUB_ENV"
+        "$(gh api repos/${{ github.repository }}/releases
+        --jq 'map(select(.draft))[0].tag_name')"
+      env:
+        GH_TOKEN: ${{ github.token }}
+    - name: Download artifacts
+      run: |
+        if [[ -z "$RELEASE_TAG" ]]; then
+          echo "Failed to find release tag" >&2
+          exit 1
+        fi
+        # For macOS, currently only x86_64 supports nested virtualization
+        # https://github.com/actions/runner-images/issues/9460
+        gh release download "$RELEASE_TAG" \
+          --repo ${{ github.repository }} \
+          --pattern '*.x86_64.dmg' \
+          --pattern '*.x86_64.dmg.sha512sum' \
+          --pattern '*.msi' \
+          --pattern '*.msi.sha512sum' \
+          --pattern 'rancher-desktop-linux-*.zip' \
+          --pattern 'rancher-desktop-linux-*.zip.sha512sum'
+      env:
+        GH_TOKEN: ${{ github.token }}
+    - name: Upload macOS artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: application-macos.zip
+        if-no-files-found: error
+        path: |
+          *.dmg
+          *.dmg.sha512sum
+    - name: Upload Windows artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: application-win32.zip
+        if-no-files-found: error
+        path: |
+          *.msi
+          *.msi.sha512sum
+    - name: Upload Linux artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: application-linux.zip
+        if-no-files-found: error
+        path: |
+          rancher-desktop-linux-*.zip
+          rancher-desktop-linux-*.zip.sha512sum
+
+  smoke-test:
+    name: Smoke test
+    needs: download-artifacts
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - { platform: macos, runs-on: macos-13, pattern: '*.aarch64.dmg' }
```

Comment:
This runs the smoke test on `aarch64` but not on `x86_64` anymore.

I also don't see any logic in the shell script that would just run the checksum/signature checks for `aarch64` but skips starting the application.